### PR TITLE
Remove smoke marker from tests

### PR DESCRIPTION
**Pull request by jgmize**
## Description

This PR eliminates the deployment pipeline step of running the smoke tests against a local instance of bedrock and a local selenium grid in docker containers on the jenkins instance. The smoke marker is still used in other situations, including running against the deployed instances.

## Issue / Bugzilla link

https://github.com/mozilla/bedrock/issues/7028

## Testing

I pushed this branch to run-integration-tests branch, triggering https://ci.vpn1.moz.works/blue/organizations/jenkins/bedrock_multibranch_pipeline/detail/run-integration-tests/293/pipeline

### CODE REVIEW

File: docs/testing.rst [hunk 1]
```diff
@@ -196,7 +196,6 @@ you can also read the `pytest markers`_ documentation for more options.
 Smoke tests
```
Comment (alexgibson) [inline docs/testing.rst]: Should we just remove this section, seeing as the smoke markers don't do anything any longer?

File: docs/pipeline.rst [hunk 2]
```diff
@@ -37,10 +37,8 @@ unit tests, as well as the smoke suite of redirect headless HTTP(s) response che
 Push to master branch
 ~~~~~~~~~~~~~~~~~~~~~
 
-Whenever a change is pushed to the master branch, the smoke suite of
-headless (see :ref:`testing-redirects`) and UI tests (see :ref:`smoke-functional-tests`)
```
Comment (alexgibson) [inline docs/pipeline.rst]: Hmm, should we still run the redirect tests at this step? I'm not clear if those will get run against a deployment or not, looking at the current pipeline?

Comment (alexgibson) [reply docs/pipeline.rst]: Could we move the redirect tests to run against a deployment instead perhaps?

File: docs/testing.rst (hunk 1 above)
Comment (jgmize) [reply docs/testing.rst]: This PR removes the machinery for running the smoke tests via local docker container, but it doesn't remove the smoke markers themselves, so the note in this section about running them from the command line still applies.

File: docs/pipeline.rst (hunk 2 above)
Comment (jgmize) [reply docs/pipeline.rst]: I was under the impression that we were already running the redirect tests against the deployments as part of the headless test suite, but I'm double checking now.

Comment (jgmize) [reply docs/pipeline.rst]: @alexgibson we are running the redirect tests against remote, but we don't currently appear to run tests with the smoke marker against remote deployments. I've pushed a new commit that should run them against the deployments in parallel to the other tests, and that's running in https://ci.vpn1.moz.works/blue/organizations/jenkins/bedrock_multibranch_pipeline/detail/run-integration-tests/296/pipeline/ now

Comment (alexgibson) [reply docs/pipeline.rst]: @jgmize at this point I really wonder what use the smoke markers are doing. Why run those only in Firefox in a separate run, and not just run all the tests for each browser?

Comment (alexgibson) [reply docs/pipeline.rst]: Also, I do see the smoke tests running against the other browser runs. Happy to chat about this more in person if it helps.

Comment (jgmize) [reply docs/pipeline.rst]: Good points @alexgibson, I agree the smoke markers aren't adding any value anymore, and I've pushed a new commit which eliminates them entirely and all references to them.

File: docs/testing.rst (hunk 1 above)
Comment (jgmize) [reply docs/testing.rst]: As per my other comment, this PR now removes the smoke markers entirely, along with this section.

**Comment by jgmize** [comment]
https://ci.vpn1.moz.works/blue/organizations/jenkins/bedrock_multibranch_pipeline/detail/run-integration-tests/297/pipeline/ is a successful run triggered by pushing https://github.com/mozilla/bedrock/commit/be619e61cabe37459073a174e683dd7cdcf04d8d to the run-integration-tests branch, and the CircleCI failure appears to have been a temporary infrastructure issue rather than one with the code; I'm rerunning it now.

**Comment by alexgibson** [comment]
@jgmize this looks great. There's a flake8 error, but aside from that I think this is good to merge 👍

**Comment by jgmize** [comment]
Thanks @alexgibson, I've pushed a new commit to fix the flake8 error.

**Comment by alexgibson** [comment]
Thanks, @jgmize 💯

**Pull request merged**